File: examples/responses_client.py
import openai

# This example demonstrates how a client sends requests to the local proxy.
# The proxy listens on http://localhost:1234 by default, then forwards to the real upstream service.
client = openai.OpenAI(
    base_url="http://localhost:1234/v1",
    api_key="sk-no-key-required",
)

# Responses API example request. Start the proxy first:
#   python -m llm_proxy --target-url http://your-upstream-url/v1
response = client.responses.create(
    model="gpt-4.1",
    instructions="You are an AI assistant. Help users with their requests.",
    input="Write a limerick about python exceptions",
)

# Print the model's final output text; full request and response will be written to logs/ by the proxy.
print(response.output_text)

File: examples/responses_client.mjs
import OpenAI from "openai";

// Install the SDK in the directory where you run this example:
//   npm install openai
// Start LLM Proxy, configure the upstream target in the web console, and enable the proxy pair.
const client = new OpenAI({
  baseURL: "http://127.0.0.1:1234/v1",
  apiKey: "sk-no-key-required",
});

const response = await client.responses.create({
  model: "gpt-4.1",
  instructions: "You are an AI assistant. Help users with their requests.",
  input: "Write a limerick about JavaScript promises.",
});

console.log(response.output_text);
